cpp/src/gandiva/decimal_type_util.h

```diff
@@ -21,13 +21,37 @@
 #define GANDIVA_DECIMAL_TYPE_UTIL_H
 
 #include <algorithm>
+#include <iostream>
 #include <memory>
+#include <string>
 
 #include "arrow/type.h"
+#include "arrow/util/decimal.h"
 #include "gandiva/arrow.h"
 
 namespace gandiva {
 
+struct DecimalLiteral {
```

Comment:
- can we make this a class, and move it to it's own file ? 
- also, name it Decimal128Full ? We'll want to use it beyond literals - for eg. in tests